### ISSUE TITLE
[openbmc] reventlog does not make too much sense

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#3497]
Looks like development was claimed to be complete in https://github.com/xcat2/xcat-core/issues/2892 so Open this as a bug. 

1) The code is not exposed to the users, but development is done, please open it up. 
```
[root@stratton01 ~]# lsxcatd -v
Version 2.13.6 (git commit 6ca241c9d0d361895d2246f410910248a1b06fbc, built Tue Jul 18 06:15:37 EDT 2017)
[root@stratton01 ~]# reventlog p9euh01
This openbmc related function is not yet supported. Please contact xCAT development team.

```

2) The format does not really make sense....

```
[root@stratton01 ~]# reventlog p9euh01
[OpenBMC development support] Using this version of xCAT, ensure firware level is at v1.99.6-0-r1, or higher.
p9euh01: 1494006358621 org.open_power.Error.Host.Event.Event
p9euh01: 1493663136573 org.open_power.Error.Host.Event.Event
p9euh01: 1493919800498 org.open_power.Error.Host.Event.Event
p9euh01: 1494606542590 org.open_power.Error.Host.Event.Event
p9euh01: 1493491950069 org.open_power.Error.Host.Event.Event
p9euh01: 1493373083349 org.open_power.Error.Host.Event.Event
p9euh01: 1495048267873 org.open_power.Error.Host.Event.Event
p9euh01: 1494097759286 org.open_power.Error.Host.Event.Event
p9euh01: 1494433643120 org.open_power.Error.Host.Event.Event
p9euh01: 1494865791307 org.open_power.Error.Host.Event.Event
p9euh01: 1493888813893 org.open_power.Error.Host.Event.Event
p9euh01: 1493828940237 org.open_power.Error.Host.Event.Event
p9euh01: 1494956796765 org.open_power.Error.Host.Event.Event
p9euh01: 1494788928045 xyz.openbmc_project.State.Host.Error.SoftOffTimeout
p9euh01: 1493576687687 org.open_power.Error.Host.Event.Event
p9euh01: 1494616195513 org.open_power.Error.Host.Event.Event
...
p9euh01: 1494606346572 org.open_power.Error.Host.Event.Event
p9euh01: 1493372853522 org.open_power.Error.Host.Event.Event
p9euh01: 1494779045502 org.open_power.Proc.CFAM.WriteFailure
p9euh01: 1495124717500 org.open_power.Proc.CFAM.ReadFailure
p9euh01: 1494443483416 org.open_power.Error.Host.Event.Event
p9euh01: 1493569745426 org.open_power.Error.Host.Event.Event
p9euh01: 1492878469134 org.open_power.Error.Host.Event.Event
p9euh01: 1492851327019 org.open_power.Error.Host.Event.Event
p9euh01: 1493569706519 org.open_power.Error.Host.Event.Event
p9euh01: 1493656187772 org.open_power.Error.Host.Event.Event
p9euh01: 1492940992240 org.open_power.Error.Host.Event.Event
p9euh01: 1493919957600 org.open_power.Error.Host.Event.Event
p9euh01: 1493310536304 org.open_power.Error.Host.Event.Event
p9euh01: 1493299597651 org.open_power.Error.Host.Event.Event
p9euh01: 1494087971882 org.open_power.Error.Host.Event.Event
p9euh01: 1494006223836 org.open_power.Error.Host.Event.Event
p9euh01: 1494779057712 org.open_power.Proc.CFAM.OpenFailure
p9euh01: 1493828842434 org.open_power.Error.Host.Event.Event
p9euh01: 1494956473709 org.open_power.Proc.CFAM.ReadFailure
p9euh01: 1494260800352 org.open_power.Error.Host.Event.Event
p9euh01: 1492800046980 org.open_power.Error.Host.Event.Event
...
```

Why is there so many Event.Event?   What is the number in the output?

**Comment by whowutwut** [comment]
@xuweibj  Are we waiting on OpenBMC team for something to implement this?  If we can close out this and expose it to help with #3652

**Comment by xuweibj** [comment]
@whowutwut , the interface for option ``clear `` is not ready. They are trying to implement. So I think maybe we could merge pull request #3533. But not support ``clear``.

**Comment by whowutwut** [comment]
@xuweibj  Yes, We should have an epic to track these sub pieces, I've reopened the reventlog epic.

**Comment by whowutwut** [comment]
Had a discussion with OpenBMC team today a little about SEL events.   The `.yaml` files in the OpenBMC project contains more human readable descriptions, https://github.com/openbmc/phosphor-dbus-interfaces/tree/master/xyz/openbmc_project  But we have not yet established how those descriptions would ship.  Need to sync up with RAS and GUI team so we are all in sync with providing consistent output. 

So digging more into IPMI, it looks like xCAT does contain a mapping between sensor codes and readable strings... so it makes sense we would have similar mapping.....   
* perl-xCAT/xCAT/data/ipmigenericevents.pm
* perl-xCAT/xCAT/data/ipmisensorevents.pm

**Comment by xuweibj** [comment]
Hi @whowutwut , shall we close this issue?

**Comment by whowutwut** [comment]
@xuweibj  No,  not yet.  There are some meeting with Justin T. who is working on RAS.  xCAT needs to understand how to make cross reference from what is provided by the logging in OpenBMC to something that is human readable.  We do this for IPMI today...  @zet809  can we take a look at how we are doing for IPMI today and understand what the BMC reports and what xCAT does for cross translation...?  Or is that translation done by `ipmitool`?

**Comment by hu-weihua** [comment]
relate to task #4352

**Comment by whowutwut** [comment]
Summary of what was done here...To improve the error messages coming back from the OpenBMC Logging API, we are re-using the mapping file provided by the `ibm-crassd` RPM for CORAL which includes a .json file that has mapping.  If the RPM is there, it would display more details about the error, if it's not there, the output will remain cryptic.  

Once the packaging is more mature on it, xCAT may ship a mapping file internally to help deal with non CORAL customers running P9, or we may just have a loose requirement of the open source .json file once we know where it is...  it's not coral specific, so p9 customers may want to have that installed anyway...  (This would reduce the need for xCAT to track this file and maintain it when things change)

**Comment by whowutwut** [comment]
To verify... using 12/13 build to verify... 

```
[root@briggs01 ~]# reventlog mid05tor12cn13 10
mid05tor12cn13: 12/05/2017 20:55:15 [33]: Firmware/Software Failure, (Critical) An internal failure has occurred while performing an operation. (AffectedSubsystem: Systems Management - Core / Virtual Appliance, PID: 1147), Resolved: 0
mid05tor12cn13: 12/06/2017 11:39:43 [34]: Firmware/Software Failure, (Warning) Unable to read the manifest file (AffectedSubsystem: Systems Management - Core / Virtual Appliance, PID: 1292), Resolved: 0
mid05tor12cn13: 12/06/2017 11:46:57 [35]: Unrecoverable Hardware Failure, (Critical) A system checkstop occurred (AffectedSubsystem: Canister/Appliance, PID: 2316), Resolved: 0
mid05tor12cn13: 12/06/2017 11:58:08 [36]: Firmware/Software Failure, (Warning) Unable to read the manifest file (AffectedSubsystem: Systems Management - Core / Virtual Appliance, PID: 1387), Resolved: 0
mid05tor12cn13: 12/06/2017 12:05:06 [37]: Unrecoverable Hardware Failure, (Critical) A system checkstop occurred (AffectedSubsystem: Canister/Appliance, PID: 4414), Resolved: 0
mid05tor12cn13: 12/06/2017 20:36:11 [38]: Firmware/Software Failure, (Warning) Unable to read the manifest file (AffectedSubsystem: Systems Management - Core / Virtual Appliance, PID: 1387), Resolved: 0
mid05tor12cn13: 12/13/2017 02:59:20 [39]: Firmware/Software Failure, (Warning) Unable to read the manifest file (AffectedSubsystem: Systems Management - Core / Virtual Appliance, PID: 1301), Resolved: 0
mid05tor12cn13: 12/13/2017 03:32:28 [40]: Firmware/Software Failure, (Warning) Unable to read the manifest file (AffectedSubsystem: Systems Management - Core / Virtual Appliance, PID: 1392), Resolved: 0
mid05tor12cn13: 12/13/2017 04:39:45 [41]: Firmware/Software Failure, (Warning) Unable to read the manifest file (AffectedSubsystem: Systems Management - Core / Virtual Appliance, PID: 1343), Resolved: 0
mid05tor12cn13: 12/13/2017 04:46:09 [42]: Firmware/Software Failure, (Warning) Unable to read the manifest file (AffectedSubsystem: Systems Management - Core / Virtual Appliance, PID: 1369), Resolved: 0
```

remove the `ibm-crassd` rpm:


```
[root@briggs01 ~]# rpm -e ibm-crassd-0.6-5.ppc64le
[root@briggs01 ~]# reventlog mid05tor12cn13 10 -s
mid05tor12cn13: Error: Only one option is supported at the same time for reventlog
[root@briggs01 ~]# reventlog mid05tor12cn13 10
mid05tor12cn13: 12/05/2017 20:55:15 [33]: xyz.openbmc_project.Common.Error.InternalFailure (PID: 1147), Resolved: 0
mid05tor12cn13: 12/06/2017 11:39:43 [34]: xyz.openbmc_project.Software.Version.Error.ManifestFileFailure (PID: 1292), Resolved: 0
mid05tor12cn13: 12/06/2017 11:46:57 [35]: org.open_power.Host.Boot.Error.Checkstop (PID: 2316), Resolved: 0
mid05tor12cn13: 12/06/2017 11:58:08 [36]: xyz.openbmc_project.Software.Version.Error.ManifestFileFailure (PID: 1387), Resolved: 0
mid05tor12cn13: 12/06/2017 12:05:06 [37]: org.open_power.Host.Boot.Error.Checkstop (PID: 4414), Resolved: 0
mid05tor12cn13: 12/06/2017 20:36:11 [38]: xyz.openbmc_project.Software.Version.Error.ManifestFileFailure (PID: 1387), Resolved: 0
mid05tor12cn13: 12/13/2017 02:59:20 [39]: xyz.openbmc_project.Software.Version.Error.ManifestFileFailure (PID: 1301), Resolved: 0
mid05tor12cn13: 12/13/2017 03:32:28 [40]: xyz.openbmc_project.Software.Version.Error.ManifestFileFailure (PID: 1392), Resolved: 0
mid05tor12cn13: 12/13/2017 04:39:45 [41]: xyz.openbmc_project.Software.Version.Error.ManifestFileFailure (PID: 1343), Resolved: 0
mid05tor12cn13: 12/13/2017 04:46:09 [42]: xyz.openbmc_project.Software.Version.Error.ManifestFileFailure (PID: 1369), Resolved: 0
Could not found '/opt/ibm/ras/lib/policyTable.json', all event messages above are original
```

I do have some comments which I will open another issue to track.

**Comment by tingtli** [comment]
Refer to https://github.com/xcat2/xcat2-task-management/issues/90
